### DISCUSSION TITLE
1.4.10: Reflow exception for an interactive e-book

### DISCUSSION THREAD

**Discussion by mikahayk** [w3c/wcag#3846] [Q&A]
The web e-book I am building has a two-dimensional layout (book pages) built on HTML5 canvas that contains interactive elements such as paragraphs, words, illustrations etc. 
It was purposefully designed to be navigated as a book, where you can leaf through the pages like a physical book. So, I found shrinking the content to 320px irrelevant in this case. 

I am wondering if this falls under Reflow exception ? 
IF NO - The product is built for Desktop usage, for Mobile/Tablet use there is a separate app available in epub format.  I can link it from the default website. Can this be considered as alternative conforming version as in [this case](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html#responsive-web-design-and-other-ways-to-meet-this-success-criterion) ?

**Comment by patrickhlauke** [comment]
> I found shrinking the content to 320px irrelevant

you may find it irrelevant, but a low vision user who may need to zoom their browser until it reaches an effective width of 320 CSS px may not...

> The product is built for Desktop usage, for Mobile/Tablet use there is a separate app available in epub format

1.4.10 isn't about mobile/tablet. it applies to desktop, and aims to ensure that a low vision user can run at a lower resolution desktop, and combine with browser zoom, to magnify things without then needing to scroll both horiztonally and vertically to read/consume individual lines of text.

**Comment by Veyfeyken** [comment]
Web is not print. Flexibility of content is of great importance to a lot of users. I'm afraid that trying to imitate the experience of reading a psychical book will work against you on many levels. For both accessibility and usability.

**Comment by bruce-usab** [comment]
Maybe I am reading between the lines, but maybe @mikahayk is building a children's puzzle book?  It is tough to make early educational materials that are fully conforming.  But, of course, there are kids with low vision.

**Comment by alastc** [comment]
(Chair hat off)

> interactive elements such as paragraphs, words, illustrations

I'm not sure how paragraphs and words are interactive, unless you mean the whole page is animated?

If it is like a comic where the text (speech bubbles?) are in a particular location on top of a picture background with characters, and reflowing those would destroy the meaning, then yes it probably would come under the 2D exception.

If you are linking through to the ePub version then that could be an alternative conforming version (assuming it did conform). 

However, I would encourage you to think about how it might reflow. For example, speech bubbles with names appearing at the top, the picture in the middle (fit to width), and the narration at the bottom. (That's how my kid's books do it.)

**Comment by mattgarrish** [comment]
> If it is like a comic where the text (speech bubbles?) are in a particular location on top of a picture background with characters, and reflowing those would destroy the meaning, then yes it probably would come under the 2D exception.

This might be getting off topic, but I was about to open an issue to try and understand exactly this.

We're currently trying to understand what parts of WCAG these kinds of fixed layout publications can meet and which they are likely to fail.

My question is whether the transform requirement applies to an image in a fixed layout page such as you'd find in a comic or manga, regardless of whether the text is character data on top or burned into the image? 

It seems like the only requirement on the image text in these cases is that it can be zoomed to 200%, whether you can find/read it without scrolling or not.

Or maybe more to the point, how do you define when the 2D presentation in an image is integral, as fixed layout pages in EPUB can also be text based, which, on the other hand, would appear to fail this SC (there's currently no way to reflow fixed layout pages). Are plain old images of text exempt when their presentation cannot be achieved any other way?

**Comment by fstrr** [comment]
This issue is labelled as a discussion, so we’re moving this to Discussions. There doesn’t seem to be an update to make to the documentation, but if that changes, we can move it back to the issues list.
